extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/dart/controller/http/DartSqlResource.java

```diff
@@ -0,0 +1,233 @@
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@Path(DartSqlResource.PATH + '/')
+public class DartSqlResource extends SqlResource
+{
+  public static final String PATH = "/druid/v2/sql/dart";
+
+  private static final Logger log = new Logger(DartSqlResource.class);
+
+  private final DartControllerRegistry controllerRegistry;
+  private final SqlLifecycleManager sqlLifecycleManager;
+  private final DartSqlClients sqlClients;
+  private final AuthorizerMapper authorizerMapper;
+  private final ResourcePermissionMapper permissionMapper;
+  private final DefaultQueryConfig dartQueryConfig;
+
+  @Inject
+  public DartSqlResource(
+      final ObjectMapper jsonMapper,
+      final AuthorizerMapper authorizerMapper,
+      @Dart final SqlStatementFactory sqlStatementFactory,
+      final DartControllerRegistry controllerRegistry,
+      final SqlLifecycleManager sqlLifecycleManager,
+      final DartSqlClients sqlClients,
+      final ServerConfig serverConfig,
+      final ResponseContextConfig responseContextConfig,
+      @Dart final ResourcePermissionMapper permissionMapper,
+      @Self final DruidNode selfNode,
+      @Dart final DefaultQueryConfig dartQueryConfig
+  )
+  {
+    super(
+        jsonMapper,
+        authorizerMapper,
+        sqlStatementFactory,
+        sqlLifecycleManager,
+        serverConfig,
+        responseContextConfig,
+        selfNode
+    );
+    this.controllerRegistry = controllerRegistry;
+    this.sqlLifecycleManager = sqlLifecycleManager;
+    this.sqlClients = sqlClients;
+    this.authorizerMapper = authorizerMapper;
+    this.permissionMapper = permissionMapper;
+    this.dartQueryConfig = dartQueryConfig;
+  }
+
+  /**
+   * API that allows callers to check if this resource is installed without actually issuing a query. If installed,
+   * this call returns 200 OK. If not installed, callers get 404 Not Found.
+   */
+  @GET
+  @Path("/enabled")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response doGetEnabled(@Context final HttpServletRequest request)
+  {
+    AuthorizationUtils.setRequestAuthorizationAttributeIfNeeded(request);
+    return Response.ok(ImmutableMap.of("enabled", true)).build();
+  }
+
+  /**
+   * API to list all running queries.
+   *
+   * @param selfOnly if true, return queries running on this server. If false, return queries running on all servers.
+   * @param req      http request
+   */
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  public GetQueriesResponse doGetRunningQueries(
+      @QueryParam("selfOnly") final String selfOnly,
```

Comment:
I went with `String` since that way the API can be `/druid/v2/sql/dart/?selfOnly` (in this case a string would be set to `""`, so we can check for non-null string). There might be a cleaner way to do this, I am not sure, but this approach is used elsewhere in the codebase. If you know a better way please LMK.